### PR TITLE
Edit Crash details: `investigator_narrative` & `fhe_collsn_id` fields

### PR DESCRIPTION
**Pull request by mateoclarke**
This is be the first PR of many. Trying to set a pattern here with how we configure our fields so that we define each of their human readable labels, whether it is editable, what UI input type we use for editing (ex: free text vs dropdown select), and what lookup table the value/ID references so we can display the human-readable description instead of an ID.

This PR only effects the `investigator_narrative` & `fhe_collsn_id` fields. More to come...

I recommend testing on Crash ID = 123
https://localhost:3000/#/crashes/123

Sep-30-2019 12-38-04

Related to #86

### CODE REVIEW

File: atd-vze/src/queries/crashes.js [hunk 1]
```diff
@@ -165,3 +167,105 @@ export const UPDATE_COORDS = gql`
     }
   }
 `;
+
+export const UPDATE_CRASH = gql`
+  mutation update_atd_txdot_crashes(
```
Comment (mateoclarke) [inline atd-vze/src/queries/crashes.js]: @sergiogcx here's the example of the mutation for editing the fields...

File: atd-vze/src/views/Crashes/Crash.js [hunk 2]
```diff
@@ -55,6 +78,35 @@ function Crash(props) {
...
+  const handleCancelClick = e => {
+    e.preventDefault();
+
+    setEditField("");
+  };
+
+  const handleFieldUpdate = e => {
+    e.preventDefault();
+
+    props.client
```
Comment (mateoclarke) [inline atd-vze/src/views/Crashes/Crash.js]: @sergiogcx, this is an important part too

File: atd-vze/src/views/Crashes/Crash.js [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
+  Col,
+  Row,
+  Table,
+  Alert,
+  Input,
+} from "reactstrap";
 import { withApollo } from "react-apollo";
 import { useQuery } from "@apollo/react-hooks";
-import crashDataMap from "./crashDataMap";
+import { crashDataMap, geoFields } from "./crashDataMap";
```
Comment (sergiogcx) [inline atd-vze/src/views/Crashes/Crash.js]: It appears neither setState or geoFields is being used, once removed this should be good to ship.

**Comment by mddilley** [comment]
I'm still looking through the changes, but, when I tested locally, I encountered this error that happens now and again. This is an awesome feature that looks and works great so this bug could probably be captured in another issue. I can reproduce by looking at any crash record details view and then refreshing the page.

Screen Shot 2019-10-02 at 11 15 36 AM

**Comment by sergiogcx** [comment]
> I'm still looking through the changes, but, when I tested locally, I encountered this error that happens now and again. This is an awesome feature that looks and works great so this bug could probably be captured in another issue. I can reproduce by looking at any crash record details view and then refreshing the page.
> 
> Screen Shot 2019-10-02 at 11 15 36 AM

I'm unable to replicate the issue even after following the same steps, tested on both firefox and chrome.

File: atd-vze/src/views/Crashes/crashDataMap.js [hunk 4]
```diff
@@ -1,104 +1,328 @@
...
+        editable: false,
+      },
+      day_of_week: {
+        label: "Day of Week",
+        editable: false,
+      },
+      fhe_collsn_id: {
+        label: "Manner of Collision ID",
+        editable: true,
+        uiType: "select",
```
Comment (mddilley) [inline atd-vze/src/views/Crashes/crashDataMap.js]: I like this pattern a lot - especially setting the uiType here and then handling it in the component. 🙌

File: atd-vze/src/views/Crashes/Crash.js (hunk 3 above)
Comment (mateoclarke) [reply atd-vze/src/views/Crashes/Crash.js]: fixed in another branch

**Pull request merged**